client/lib/signup/step-actions/index.js

```diff
@@ -1246,29 +1247,13 @@ export function maybeAddStorageAddonToCart( stepName, defaultDependencies, nextP
 	const state = store.getState();
 	const selectedStorage = get( getSignupDependencyStore( state ), 'storage', null );
 
-	switch ( selectedStorage ) {
-		case AddOns.ADD_ON_50GB_STORAGE:
-			cartItem.push( {
-				product_slug: PRODUCT_1GB_SPACE,
-				quantity: 50,
-				volume: 1,
-				extra: { feature_slug: AddOns.ADD_ON_50GB_STORAGE },
-			} );
-			recordTracksEvent( 'calypso_signup_storage_add_on_selected', {
-				add_on_slug: AddOns.ADD_ON_50GB_STORAGE,
-			} );
-			break;
-		case AddOns.ADD_ON_100GB_STORAGE:
-			cartItem.push( {
-				product_slug: PRODUCT_1GB_SPACE,
-				quantity: 100,
-				volume: 1,
-				extra: { feature_slug: AddOns.ADD_ON_100GB_STORAGE },
-			} );
-			recordTracksEvent( 'calypso_signup_storage_add_on_selected', {
-				add_on_slug: AddOns.ADD_ON_100GB_STORAGE,
-			} );
-			break;
+	if ( STORAGE_ADD_ONS.includes( selectedStorage ) ) {
```

Comment:
The add on slug -> quantity mapping is already defined here: https://github.com/Automattic/wp-calypso/blob/e0a01efcf7c7f5305ed0be93a48524d67f50fa0c/packages/data-stores/src/add-ons/add-ons-list.ts#L26

Can we use that instead of redefining it? Also, can we preserve the Tracks event? 🙂

Reply:
Nice catch on the event and I will use the tracks event